### ISSUE TITLE
WCAG version 2.0

### ISSUE THREAD

**Issue by jladbury** [bgrins/TinyColor#76]
This is an excellent library! Thank you.

Reading the annotated source code, I see that the readability routines refer to http://www.w3.org/TR/AERT#color-contrast, which in turn is based on the  Web Content Accessibility Guidelines Version 1.0.

I have previously incorporated the slightly different WCAG Version 2.0 algorithm (http://www.w3.org/TR/2008/REC-WCAG20-20081211/#contrast-ratiodef) in a VBA project, and will probably update my downloaded copy of this library to use it. I am unfamiliar with working in the Open Source world, but if someone could give me a brief description of what I would need to do to port that work back (assuming anyone is interested!) I would be happy to contribute.

**Comment by bgrins** [comment]
@jladbury yes, I'd be happy to accept a pull request with the updated accessibility guidelines.  There is a guide for how to send this change over using github here: https://help.github.com/articles/using-pull-requests/.

For this change, in addition to updating the code in tinycolor.js, you will need to also update the tests [here](https://github.com/bgrins/TinyColor/blob/c32e7b068c63f5b76742c79e9f53a2a2c2bd79ba/test/test.js#L552).

**Comment by jladbury** [comment]
@bgrins thanks for a quick response.  I have so far coded the changes so that they hopefully will not break anyone's existing use. Briefly, I have:
1.  added a new function **getLuminance** modelled on **getBrightness**, but using the WCAG 2.0 algorithm:
2. updated the **readability** function to use the above to add a **contrast** field to the returned object.

Yes, I noticed the tests - they already proved excellent for regression testing of my changes so far.I have some test cases from my VBA project, and will incorporate these and probably a few more. Regarding that, I noticed that  the *Utilities / readability* test module has a comment about *Need tests for readability*. Am I right in thinking that it will be appropriate to replace the *deepequal* assertions with some actual examples that accord to the guidelines, both WCAG 1.0 and 2.0?

**Comment by bgrins** [comment]
> Am I right in thinking that it will be appropriate to replace the deepequal assertions with some actual examples that accord to the guidelines, both WCAG 1.0 and 2.0?

Yes, if you have some additional test cases I would love to expand that coverage to make sure that we are actually following the spec.

**Comment by jladbury** [comment]
@bgrins,
Brian , I am about to submit my first Pull request. Never done one before, so I hope it works.
And then of course I hope you like the work!
